src/typing.py

```diff
@@ -1221,9 +1221,9 @@ def __getitem__(self, parameters):
         elif args == []:
             parameters = ((), result)
         else:
-            if not isinstance(args, list):
-                raise TypeError("Callable[args, result]: args must be a list."
-                                " Got %.100r." % (args,))
+            if not isinstance(args, list) or ... in args or () in args:
```

Comment:
@gvanrossum Yes, indeed I wanted to flatten ``__parameters__`` to just call ``super().__getitem__``. But you are right, it is not necessary to do this here, but only right before I actually call ``super()``.